### PR TITLE
feat: loan card activity feed on filters page

### PR DESCRIPTION
**Pull request by roger-in-kiva**
- Activity feed showed in loan cards on team picks filter page

Screenshot 2024-04-02 at 21 02 50

Screenshot 2024-04-02 at 21 02 58

### CODE REVIEW

File: src/components/LoanCards/KvClassicLoanCardContainer.vue [hunk 1]
```diff
@@ -274,7 +285,68 @@ export default {
 		},
 		setCookie(name, value, options) {
 			return this.cookieStore.set(name, value, options);
-		}
+		},
+		async fetchLoanActivity() {
+			const activities = [];
+
+			const activityResponse = await this.apollo.query({
```
Comment (dyersituations) [inline src/components/LoanCards/KvClassicLoanCardContainer.vue:292]: There should be a try catch around this query

File: src/components/LoanCards/KvClassicLoanCardContainer.vue [hunk 2]
```diff
@@ -274,7 +285,68 @@ export default {
...
+
+			lendingActionsData.forEach(action => {
+				const actionDate = new Date(action?.latestSharePurchaseDate).toDateString();
+				if (!activities.some(activity => activity.key === actionDate)) {
+					activities.push({
+						key: actionDate,
+						data: [],
+					});
+				}
+				if (action?.lender.name) {
```
Comment (dyersituations) [inline src/components/LoanCards/KvClassicLoanCardContainer.vue:308]: Is optional chaining needed here? `action?.lender?.name`

File: src/components/LoanCards/KvClassicLoanCardContainer.vue [hunk 3]
```diff
@@ -274,7 +285,68 @@ export default {
...
+					activities.push({
+						key: commentDate,
+						data: [],
+					});
+				}
+				if (comment?.authorName) {
+					const dataObject = activities.find(activity => activity.key === commentDate);
+					dataObject?.data.push({
+						lenderName: comment.authorName,
+						lenderImage: comment.authorLendingAction?.lender.image?.url,
```
Comment (dyersituations) [inline src/components/LoanCards/KvClassicLoanCardContainer.vue:332]: Looks like this needs one more chaining: `comment.authorLendingAction?.lender?.image?.url`

File: src/components/LoanCards/KvClassicLoanCardContainer.vue (hunk 2 above)
Comment (roger-in-kiva) [reply src/components/LoanCards/KvClassicLoanCardContainer.vue]: There is no activity w/o lender data, but good practice.

**Review by dyersituations — APPROVED** [review]

**Review by mcstover — APPROVED** [review]
Neat, this is probably going to show up in some corporate campaign loan views soon too. We'll prob need to add some addition configs as I don't think we're going to show the pop option

**Pull request merged**